gnmi_server/gnoi_system_test.go

```diff
@@ -0,0 +1,261 @@
+package gnmi
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/sonic-net/sonic-gnmi/common_utils"
+
+	syspb "github.com/openconfig/gnoi/system"
+	"github.com/openconfig/gnoi/types"
+	"github.com/redis/go-redis/v9"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+)
+
+func errorCodeToSwss(errCode codes.Code) string {
+	switch errCode {
+	case codes.OK:
+		return "SWSS_RC_SUCCESS"
+	case codes.Unknown:
+		return "SWSS_RC_UNKNOWN"
+	case codes.InvalidArgument:
+		return "SWSS_RC_INVALID_PARAM"
+	case codes.DeadlineExceeded:
+		return "SWSS_RC_DEADLINE_EXCEEDED"
+	case codes.NotFound:
+		return "SWSS_RC_NOT_FOUND"
+	case codes.AlreadyExists:
+		return "SWSS_RC_EXISTS"
+	case codes.PermissionDenied:
+		return "SWSS_RC_PERMISSION_DENIED"
+	case codes.ResourceExhausted:
+		return "SWSS_RC_FULL"
+	case codes.Unimplemented:
+		return "SWSS_RC_UNIMPLEMENTED"
+	case codes.Internal:
+		return "SWSS_RC_INTERNAL"
+	case codes.FailedPrecondition:
+		return "SWSS_RC_FAILED_PRECONDITION"
+	case codes.Unavailable:
+		return "SWSS_RC_UNAVAIL"
+	}
+	return ""
+}
+
+func nsfManagerResponse(t *testing.T, sc *redis.Client, expectedResponse codes.Code, fvs map[string]string, done chan bool, key string) {
+	sub := sc.Subscribe(context.Background(), "Reboot_Request_Channel")
+	if _, err := sub.Receive(context.Background()); err != nil {
+		t.Errorf("nsfManagerResponse failed to subscribe to request channel: %v", err)
+		return
+	}
+	defer sub.Close()
+	channel := sub.Channel()
+
+	np, err := common_utils.NewNotificationProducer("Reboot_Response_Channel")
+	if err != nil {
+		t.Errorf("nsfManagerResponse failed to create notification producer: %v", err)
+		return
+	}
+	defer np.Close()
+
+	tc := time.After(5 * time.Second)
+	select {
+	case msg := <-channel:
+		t.Logf("nsfManagerResponse received request: %v", msg)
+		// Respond to the request
+		if err := np.Send(key, errorCodeToSwss(expectedResponse), fvs); err != nil {
+			t.Errorf("nsfManagerResponse failed to send response: %v", err)
+			return
+		}
+	case <-done:
+		return
+	case <-tc:
+		t.Error("nsfManagerResponse timed out waiting for request")
+		return
+	}
+}
+
+func TestGnoiSystem(t *testing.T) {
```

Comment:
Do we have any testing for subcomponents field?

Reply:
There's no specific test for the subcomponents field, will update to add one for the subcomponents.